test/functorch/test_control_flow.py

```diff
@@ -1208,6 +1207,7 @@ def g(pred, xs, y):
         self.assertEqual(res, g(pred, x, y))
         self.check_map_count(gm, 1)
 
+    @unittest.skipIf(TEST_WITH_TORCHDYNAMO, "Require making map and map_impl to have the same signature.")
```

Comment:
Test with dynamo makes the eager graph execution of map_impl in L1241 go through dynamo. For map_impl, right now we create a MapHigherOrder for it but when we turn on input validation in this diff, it creates a graph break, which is normally OK but since the upper-level cond requires it to be full captured, this introduces an error.